### PR TITLE
fix: instagram regex

### PR DESCRIPTION
**Pull request by nastita**
### Ticket ID
DEV-11799

### Description
Update Instagram widget parameters regex parsing

### CODE REVIEW

**Comment by Hugoo** [comment]
Task linked: [DEV-11799 Fix Instagram parser](https://app.clickup.com/t/2645698/DEV-11799)

File: domains/grid/utils/gridParser.ts [hunk 1]
```diff
@@ -135,14 +135,16 @@ const parseSoundCloudWidgetInput = (
...
   input: string
 ): LayoutItemExtended | never => {
-  // TODO: Add support for instagram timeline embeds when we have the widget type
-  const INSTAGRAM_PERMALINK_REGEX = /data-instgrm-permalink="([^"]+)"/
-  const instagramMatch = input.match(INSTAGRAM_PERMALINK_REGEX)
-  if (instagramMatch) {
+  const INSTAGRAM_URL_REGEX =
+    /https:\/\/www\.instagram\.com\/(p|reel|profile|tv)\/[\w-]+\/(?:\?[^"]*)?/
+
+  const match = input.match(INSTAGRAM_URL_REGEX)
```
Comment (dzbo) [inline domains/grid/utils/gridParser.ts:141]: Can you use spread to name match parts like I did in other parsers? I'm not a fan of using indexes as they hide the purpose of the value.

**Review by dzbo — COMMENTED** [review]
Have you tried other embeds too? There is `profile` and `tv` from what I see.

**Comment by nastita** [comment]
> Have you tried other embeds too? There is `profile` and `tv` from what I see.

@dzbo I wasn't able to find a single embeddable profile. I don't know if everyone has embed disabled for profiles or if ig just removed embedding profiles.
I don't even know how igtv works or is haha, I know it exists. For the time being we can just say something like "not supported yet" for those embed types.

**Review by dzbo — APPROVED** [review]

**Pull request merged**